File: ruby/trip/trip_dao.rb
require_relative '../еxceptions/dependend_class_call_during_unit_test_exception'

class TripDAO
    def self.find_trips_by_user(user)
        raise DependendClassCallDuringUnitTestException.new('TripDAO should not be invoked on an unit test.')
    end
end

File: TypeScript/src/trip/TripDAO.ts
import CollaboratorCallException from "../exception/CollaboratorCallException";
import User from "../user/User";
import Trip from "./Trip";

export default class TripDAO {
    public static findTripsByUser(user: User): Trip[] {
        throw new CollaboratorCallException(
            "TripDAO should not be invoked on an unit test.");
    }
}
